<?xml version="1.0"?>
<launch>
    <arg name="map_file" default="$(find rosplan_tiago_demo)/maps/tiago_office_house.yaml"/>
    <arg name="gazebo_gui" default="true"/>

<!--     <arg name="world"  default="turtlebot3_house"/>   --><!-- empty, ... (see ../worlds) -->
<!--     <arg name="world"  default="modified_small_office"/> -->
<!--     <arg name="world"  default="tiago_office_navigation_pickup"/> -->
    <arg name="world"  default="tiago_asp_task_domain"/>
<!--     <arg name="world"  default="modified_small_office_large_aruco"/> -->
    <!--deprecated argument, you should specify the parameters below-->
    <!-- They are described in tiago.urdf.xacro -->
    <arg name="robot"  default="steel"/>
    <arg name="arm"  default="$(eval {'iron': False}.get(arg('robot'), True))"/>
    <arg name="end_effector"  default="$(eval {'iron': 'false', 'steel': 'pal-gripper', 'titanium': 'pal-hey5'}.get(arg('robot'), 'pal-gripper'))"/>
    <arg name="ft_sensor"  default="$(eval {'titanium': 'schunk-ft'}.get(arg('robot'), 'false'))"/>
    <arg name="laser_model"  default="sick-571"/>
    <arg name="camera_model"  default="orbbec-astra"/>
    <arg name="multi" default=""/>
<!--     <arg name="gzpose" default="-x -3.9 -y 3.0 -z 0.0 -R 0.0 -P 0.0 -Y 0.0"/> -->
<!--     <arg name="gzpose" default="-x -1.0 -y 0.0 -z 0.0 -R 0.0 -P 0.0 -Y 0.0"/> -->
<!--     <arg name="gzpose" default="-x 4.25 -y -11.98 -z -0.00156513154922 -R 0.0 -P 0.0 -Y 0.0"/> -->
    <arg name="lost" default="false" />
    <arg name="gzpose" default="-x 0.0  -y   0.0 -z  0.0   -R 0.0 -P 0.0 -Y  0.0" unless="$(arg lost)" />
    <arg name="gzpose" default="-x 0.78 -y -11.1 -z -0.003 -R 0.0 -P 0.0 -Y 70.0" if="$(arg lost)" />
    <arg name="gui" default="true"/>
    <arg name="debug" default="false"/>
    <arg name="public_sim" default="false"/>
    <arg name="tuck_arm" default="true"/>
    <arg name="recording" default="false"/>
    <arg name="extra_gazebo_args" default=""/>
    <arg name="use_moveit_camera" default="false"/>
    <arg name="use_dynamic_footprint" default="false"/>

    <env name="LC_NUMERIC" value="C" />

    <!-- tiago robot model and world files -->

    <!-- start up world -->
    <include file="$(find pal_gazebo_worlds)/launch/pal_gazebo.launch">
        <arg name="world" value="$(arg world)"/>
        <arg name="gui" value="$(arg gui)"/>
        <arg name="debug" value="$(arg debug)"/>
        <arg name="recording" value="$(arg recording)"/>
        <arg name="extra_gz_model_path" value="$(find tiago_gazebo)/models"/>
        <arg name="extra_gz_resource_path" value="$(find tiago_gazebo)"/>
        <arg name="extra_gazebo_args" value="$(arg extra_gazebo_args)"/>
    </include>

    <!-- spawn robot in simulation -->
    <include file="$(find rosplan_tiago_demo)/launch/tiago_spawn.launch">
        <arg name="arm" value="$(arg arm)"/>
        <arg name="end_effector" value="$(arg end_effector)"/>
        <arg name="ft_sensor" value="$(arg ft_sensor)"/>
        <arg name="laser_model" value="$(arg laser_model)"/>
        <arg name="camera_model" value="$(arg camera_model)"/>
        <arg name="multi" value="$(arg multi)"/>
        <arg name="gzpose" value="$(arg gzpose)"/>
    </include>

    <!-- bring up robot -->
    <include file="$(find rosplan_tiago_demo)/launch/simulation_tiago_bringup.launch">
        <arg name="arm" value="$(arg arm)"/>
        <arg name="end_effector" value="$(arg end_effector)"/>
        <arg name="ft_sensor" value="$(arg ft_sensor)"/>
        <arg name="laser_model" value="$(arg laser_model)"/>
        <arg name="camera_model" value="$(arg camera_model)"/>
        <arg name="public_sim"  value="$(arg public_sim)"/>
        <arg name="use_moveit_camera" value="$(arg use_moveit_camera)"/>
        <arg name="use_dynamic_footprint" value="$(arg use_dynamic_footprint)"/>
    </include>

      <!-- point cloud throttle and filter -->
    <group unless="$(arg public_sim)">
    <!--     <include file="$(find pal_pcl_points_throttle_and_filter)/launch/throttle_filtering_points.launch"> -->
    <!--       <arg name="cloud"  value="/xtion/depth_registered/points"/> -->
    <!--     </include> -->
    </group>

    <group if="$(arg tuck_arm)">
        <node name="tuck_arm" pkg="tiago_gazebo" type="tuck_arm.py" output="screen"/>
    </group>
    <node name="image_proc" pkg="image_proc" type="image_proc" ns="xtion/rgb"/>

<!--    --><!-- marker detector -->
<!--   <node pkg="aruco_ros" type="single"  name="aruco_single"> -->
<!--       <remap from="/camera_info"       to="/xtion/rgb/camera_info" /> -->
<!--       <remap from="/image"             to="/xtion/rgb/image_rect_color" /> -->
<!--       <param name="image_is_rectified" value="True"/> -->
<!--       <param name="marker_size"        value="0.045"/> -->
<!--       <param name="marker_id"          value="582"/> -->
<!--       <param name="reference_frame"    value="/base_footprint"/>    --><!-- frame in which the marker pose will be expressed -->
<!--       <param name="camera_frame"       value="xtion_rgb_optical_frame"/> -->
<!--       <param name="marker_frame"       value="aruco_frame" /> -->
<!--       <param name="corner_refinement"  value="SUBPIX" /> -->
<!--   </node> -->


    <!-- rosplan framework -->
<!--     <include file="$(find rosplan_interface_mapping)/launch/rosplan_roadmap_server.launch"/> -->

<!--     <include file="$(find rosplan_planning_system)/launch/interfaced_planning_system.launch"> -->
<!--         <arg name="domain_path" value="$(find rosplan_tiago_demo)/common/domain_tiago_demo.pddl"/> -->
<!--         <arg name="problem_path" value=""/> -->
<!--         <arg name="autom_gen_problem_path" value="$(find rosplan_tiago_demo)/common/problem.pddl"/> -->
<!--         <arg name="data_path" value="$(find rosplan_tiago_demo)/common/"/> -->
<!--     </include> -->

<!--     <node pkg="rosplan_interface_movebase" type="rpmovebase" name="rosplan_interface_movebase" respawn="false" output="screen"> -->
<!--         <param name="knowledge_base" value="rosplan_knowledge_base"/> -->
<!--         <param name="action_server" value="/move_base"/> -->
<!--         <param name="pddl_action_name" value="goto_waypoint"/> -->
<!--         <param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch"/> -->
<!--         <param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback"/> -->
<!--     </node> -->

<!--     <node name="rviz" pkg="rviz" type="rviz" args="-d $(find rosplan_turtlebot3_demo)/config/turtlebot3.rviz" /> -->


<!--     <node pkg="rosplan_tiago_demo" type="pickup_action_interface.py" name="pickup_action_interface" output="screen"/> -->
<!--     <node pkg="rosplan_tiago_demo" type="sense_action_interface.py" name="sense_action_interface" -->
<!--     args="$(find rosplan_tiago_demo)/graphs/graph_wg.xml $(find rosplan_tiago_demo)/graphs/angle_wg.xml $(find rosplan_tiago_demo)/graphs/vps_wg.xml" output="screen"/> -->

    <node pkg="web_video_server" type="web_video_server" name="web_video_server"/>
</launch>
